mne/epochs.py

```diff
@@ -183,6 +184,93 @@ def _save_split(epochs, fname, part_idx, n_parts, fmt):
     end_file(fid)
 
 
+def _handle_duplicate_events(events, event_id, event_repeated):
+    """Handle duplicated events."""
+    u_evs, u_idxs, counts = np.unique(events[:, 0], return_index=True,
+                                      return_counts=True)
+
+    # Return early if no duplicates
+    if len(u_evs) == len(events):
+        return events, event_id
+
+    # Else, we have duplicates. Triage ...
+    event_id = event_id.copy()
+    msg = 'Multiple event codes for single event times found.'
+    if event_repeated == 'error':
+        raise RuntimeError('Event time samples were not unique')
+
+    elif event_repeated == 'drop':
+        logger.info('{} Keeping the first occurrence and dropping all others.'
+                    .format(msg))
+        new_events = events[u_idxs]
+
+    elif event_repeated == 'merge':
+        logger.info('{} Creating new event code to reflect simultaneous '
+                    'events and updating event_id.'.format(msg))
+        new_events = events.copy()
+        non_u_evs = u_evs[counts > 1]
+        for ev in non_u_evs:
+
+            # indices at which the non-unique events happened
+            idxs = (events[:, 0] == ev).nonzero()[0]
+
+            # Figure out event_codes prior to events to be merged, fall back
+            # to 0, if heterogeneous codes exist
+            prior_codes = events[idxs, 1]
+            new_prior = 0
+            if len(np.unique(prior_codes)) == 1:
+                new_prior = np.unique(prior_codes)[0]
+
+            # Make an event_id for the merged event
+            ev_codes = events[idxs, 2]
+
+            if len(np.unique(ev_codes)) > 1:
+
+                new_event_id_key = list()
+                for code in ev_codes:
+                    # inverse dict lookup, because event_id is a one-to-one map
+                    kk = list(event_id.keys())[list(event_id.values())
+                                               .index(code)]
+                    new_event_id_key.append(kk)
+                new_event_id_key = '/'.join(new_event_id_key)
+
+                # Check if we already have a corresponding code
+                new_event_code = event_id.get(new_event_id_key, False)
```

Comment:
this is weird. `new_event_code` its either a valid id_key or a different type (bool equal to False)??

Reply:
I agree, I use a boolean flag now that is always of type bool. Thanks!